libhb/fifo.c

```diff
@@ -674,6 +677,29 @@ void hb_buffer_close( hb_buffer_t ** _b )
 {
     hb_buffer_t * b = *_b;
 
+#ifdef USE_QSV
+    // Reclaim QSV resources before dropping the buffer.
+    // when decoding without QSV, the QSV atom will be NULL.
+    if (b->qsv_details.qsv_atom != NULL && b->qsv_details.ctx != NULL)
```

Comment:
just from the short run test - I would do:
if (b !=NULL && b->qsv_details.qsv_atom != NULL && b->qsv_details.ctx != NULL)